src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoSearchIT.java

```diff
@@ -843,13 +843,46 @@ public void searchTransactionsByCursor() {
                 .withCreatedDate(now(ZoneOffset.UTC).minusDays(1))
                 .insert(rule.getJdbi());
 
-        searchParams.setAccountId("1");
+        searchParams.setAccountIds("1");
 
         List<TransactionEntity> transactionList = transactionDao.cursorTransactionSearch(searchParams, null, null);
 
         assertThat(transactionList.size(), is(2));
     }
 
+    @Test
+    public void searchTransactionsByMultipleGatewayAccounts() {
+        aTransactionFixture()
+                .withGatewayAccountId("1")
+                .withCreatedDate(now(ZoneOffset.UTC).minusDays(1))
+                .insert(rule.getJdbi());
+
+        aTransactionFixture()
+                .withGatewayAccountId("1")
+                .withCreatedDate(now(ZoneOffset.UTC).minusDays(1))
+                .insert(rule.getJdbi());
+
+        aTransactionFixture()
+                .withGatewayAccountId("2")
+                .withCreatedDate(now(ZoneOffset.UTC).minusDays(1))
+                .insert(rule.getJdbi());
+
+        aTransactionFixture()
+                .withGatewayAccountId("1")
+                .withCreatedDate(now(ZoneOffset.UTC).minusDays(1))
+                .insert(rule.getJdbi());
+
+        aTransactionFixture()
+                .withGatewayAccountId("1337")
+                .withCreatedDate(now(ZoneOffset.UTC).minusDays(1))
+                .insert(rule.getJdbi());
+
+        searchParams.setAccountIds(List.of("1","2"));
+
+        List<TransactionEntity> transactionList = transactionDao.searchTransactions(searchParams);
+
+        assertThat(transactionList.size(), is(4));
```

Comment:
Might be worth sticking some sort of unique identifier (e.g. an external ID) on each transaction so we can verify we’re getting the right ones.